src/lib.rs

```diff
@@ -97,6 +96,7 @@ pub enum Error {
     /// The operation involved a conversion to Rust's native String type, which failed because the
     /// string did not contain all valid UTF-8.
     InvalidUtf8,
+    UnsupportedOperation,
```

Comment:
This needs a doccomment. It should make it clear that this isn't a `libc` error, but a `nix`-specific error indicating that the operation is not supported.